### PR TITLE
add StakingEvent

### PR DESCRIPTION
**Pull request by jeremyletang**
Add the staking event required for the staking.Accounting engine

### CODE REVIEW

File: sources/vega/events/v1/events.proto [hunk 1]
```diff
@@ -12,7 +12,31 @@ import "vega/oracles/v1/data.proto";
 import "vega/commands/v1/commands.proto";
 import "vega/commands/v1/validator_commands.proto";
 
-//DelegationBalanceEvent - updates on the delegation balance of a party to a node in the current epoch in effect 
+// StakingEvent - an event notifying of stake being deposited or removed for a given party
+// These events are emitted for every Staking deposit or removed accepted by the network
+message StakingEvent {
+  enum Kind {
+    // Default value
+    KIND_UNSPECIFIED = 0;
```
Comment (EVODelavega) [inline sources/vega/events/v1/events.proto:20]: Is this another one of those kind vs type things, or can we use `Type` instead of `Kind` here?

**Review by peterbarrow — APPROVED** [review]

File: sources/vega/events/v1/events.proto [hunk 2]
```diff
@@ -12,7 +12,31 @@ import "vega/oracles/v1/data.proto";
...
+    // Indicate of a stake deposit instruction
+    KIND_DEPOSIT = 1;
+    // Indicate of a stake remove instruction
+    KIND_REMOVE = 2;
```
Comment (ze97286) [inline sources/vega/events/v1/events.proto:24]: WITHDRAW?

**Review by ze97286 — APPROVED** [review]

File: sources/vega/events/v1/events.proto (hunk 2 above)
Comment (jeremyletang) [reply sources/vega/events/v1/events.proto:24]: Well, the ethereum event have been called Remove, so I matched the naming with it there: https://github.com/vegaprotocol/Staking_Bridge/blob/main/contracts/IStake.sol#L10.

Do you want me to name it withdraw here in the proto? I'm happy to.

File: sources/vega/events/v1/events.proto (hunk 1 above)
Comment (jeremyletang) [reply sources/vega/events/v1/events.proto:20]: We can if you prefer, I'm not fussy about it.

Comment (jeremyletang) [reply sources/vega/events/v1/events.proto]: addressed.

File: sources/vega/events/v1/events.proto (hunk 2 above)
Comment (ze97286) [reply sources/vega/events/v1/events.proto]: leave it for now

**Pull request merged**